pandas/core/dtypes/base.py

```diff
@@ -276,10 +276,12 @@ def is_dtype(cls, dtype) -> bool:
             return False
         elif isinstance(dtype, cls):
             return True
-        try:
-            return cls.construct_from_string(dtype) is not None
-        except TypeError:
-            return False
+        if isinstance(dtype, str):
+            try:
+                return cls.construct_from_string(dtype) is not None
+            except TypeError:
+                return False
```

Comment:
is it still possible to get a TypeError here?  if so, should it be?

Reply:
I think so. A 3rd-party EA may have overridden `construct_from_string`, and we document that it should raise a TypeError if you can't construct it from that string.